legend-engine-authentication-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/auth/IntermediationRule.java

```diff
@@ -0,0 +1,46 @@
+package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.auth;
+
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.AuthenticationSpec;
+import org.finos.legend.engine.shared.core.identity.Credential;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public abstract class IntermediationRule<Spec, InputCred extends Credential, OutputCred extends Credential>
```

Comment:
shouldn't intermediationRule be about going from 1 cred to another? why is Spec here? 
Seems to be violation of layers?